rotkehlchen/tests/api/test_blockchain.py

```diff
@@ -1877,107 +1667,59 @@ def test_add_ksm_blockchain_account_invalid(rotkehlchen_api_server):
-            },
-        )
-        if async_query:
-            task_id = assert_ok_async_response(response)
-            result = wait_for_async_task_with_result(rotkehlchen_api_server, task_id)
-        else:
-            result = assert_proper_response_with_result(response)
-
-    # Check per account
-    account_balances = result['per_account']['KSM'][SUBSTRATE_ACC1_KSM_ADDR]
-    assert 'liabilities' in account_balances
-    asset_ksm = account_balances['assets']['KSM']
-    assert FVal(asset_ksm['amount']) >= ZERO
-    assert FVal(asset_ksm['usd_value']) >= ZERO
+    if async_query:
+        task_id = assert_ok_async_response(response)
+        wait_for_async_task_with_result(rotkehlchen_api_server, task_id)
+    else:
+        assert_proper_response_with_result(response)
 
-    # Check totals
-    assert 'liabilities' in result['totals']
-    total_ksm = result['totals']['assets']['KSM']
-    assert FVal(total_ksm['amount']) >= ZERO
-    assert FVal(total_ksm['usd_value']) >= ZERO
+    assert rotki.chain_manager.accounts.ksm == [SUBSTRATE_ACC1_KSM_ADDR]
 
 
 @pytest.mark.parametrize('number_of_eth_accounts', [0])
 @pytest.mark.parametrize('ksm_accounts', [[SUBSTRATE_ACC1_KSM_ADDR, SUBSTRATE_ACC2_KSM_ADDR]])
-@pytest.mark.parametrize('kusama_manager_connect_at_start', [KUSAMA_TEST_NODES])
 def test_remove_ksm_blockchain_account(rotkehlchen_api_server):
     """Test removing a Kusama blockchain account works as expected by returning
     only the balances of the other Kusama accounts.
     """
     rotki = rotkehlchen_api_server.rest_api.rotkehlchen
     async_query = random.choice([False, True])
-
-    setup = setup_balances(
-        rotki=rotki,
-        ethereum_accounts=None,
-        btc_accounts=None,
-        eth_balances=None,
-        token_balances=None,
-        btc_balances=None,
+    response = requests.delete(
+        api_url_for(
+            rotkehlchen_api_server,
+            "blockchainsaccountsresource",
+            blockchain=SupportedBlockchain.KUSAMA.value,
+        ),
+        json={
+            'accounts': [SUBSTRATE_ACC2_KSM_ADDR],
+            'async_query': async_query,
+        },
     )
-    with ExitStack() as stack:
-        setup.enter_blockchain_patches(stack)
-        response = requests.delete(
-            api_url_for(
-                rotkehlchen_api_server,
-                "blockchainsaccountsresource",
-                blockchain=SupportedBlockchain.KUSAMA.value,
-            ),
-            json={
-                'accounts': [SUBSTRATE_ACC2_KSM_ADDR],
-                'async_query': async_query,
-            },
-        )
-        if async_query:
-            task_id = assert_ok_async_response(response)
-            result = wait_for_async_task_with_result(rotkehlchen_api_server, task_id)
-        else:
-            result = assert_proper_response_with_result(response)
-
-    # Check per account
-    assert SUBSTRATE_ACC2_KSM_ADDR not in result['per_account']['KSM']
```

Comment:
It would be useful I think to see that the totals are updated. But to do that properly you would need to modify the test.

you would need to add (1) a query at the start to have balances populated, then (2) delete the accounts as is already done and then (3) check totals only -- no need to query, to see totals are updated without a new api call.